boards/nxp/hexiwear/hexiwear_mk64f12.dts

```diff
@@ -186,24 +186,20 @@
 			reg = <0x00000000 0x00010000>;
 			read-only;
 		};
-
-		/*
-		 * The flash starting at 0x00010000 and ending at
-		 * 0x0001ffff (sectors 16-31) is reserved for use
-		 * by the application.
+		/* Note slot 0 has one additional sector,
+		 * this is intended for use with the swap move algorithm
 		 */
-
-		slot0_partition: partition@20000 {
+		slot0_partition: partition@10000 {
 			label = "image-0";
-			reg = <0x00020000 0x00060000>;
+			reg = <0x00010000 0x00069000>;
 		};
-		slot1_partition: partition@80000 {
+		slot1_partition: partition@79000 {
 			label = "image-1";
-			reg = <0x00080000 0x00060000>;
+			reg = <0x00079000 0x00068000>;
```

Comment:
Only one sector different that I can see here (assuming 4KiB sector size), this does not seem right. If you build MCUboot with logging enabled and MCUboot log level set to debug, if the sector size here is not optimal then you should see a debug log message about it, have you checked these sizes?

Reply:
> Only one sector different that I can see here (assuming 4KiB sector size), this does not seem right. If you build MCUboot with logging enabled and MCUboot log level set to debug, if the sector size here is not optimal then you should see a debug log message about it, have you checked these sizes?

Yes. The sector size for K64 is 4KiB = 0x1000. 
- "Slot0 size" set to 0x00069000
- "Slot1 size" set to 0x00068000

"Slot 0 size"  > "Slot 1 size" by 0x1000 = 4KBs. 
So it is fine.

Reply:
That is not optimal for swap using move, as said

Reply:
> That is not optimal for swap using move, as said

Please read the MCUBoot documentation:
https://docs.mcuboot.com/design.html#image-swap-no-scratch

**" Therefore the most memory-size-effective slot layout is when the primary slot is exactly one sector larger than the secondary slot"**

Reply:
That applies to the move sector only, it does not include the swap status data, if you run the code with debug logging you will see a log message clearly explaining that the sector layout here is not optimal as I keep saying, your slot1 has at least 1 completely wasted sector that can never be used in this setup here.

Reply:
Do not understand.
Do you mean that Slot 1 size must be equal to Slot 0?
Or could you specify what size of Slot 0 and Slot 2 should be?
Could you point o the documentation or source code?

Reply:
https://docs.mcuboot.com/design.html#image-trailer is the documentation for it, MCUboot needs to store what sectors have been swapped, so you need one sector for all sectors to be moved up by, then you need at least one sector (it could be more, it depends on the value of the Kconfig specifying how many sectors a build will support) to store the status for each sector when they get swapped between each image - this data goes in the primary slot.

The reason why I keep saying build MCUboot with logging and MCUboot log level set to debug is that there is a specific message that will be printed if the number of sectors assigned to your slots has an issue and basically tells you the number of sectors that would work for your configuration. For example:
```
I: Starting bootloader
D: Non-optimal sector distribution, slot0 has 62 usable sectors (64 assigned) but slot1 has 64 assigned
```

Reply:
As the board is not available, not able to get the log.
Later, if somebody will find this board, we can adjust its value to make it more optimal.

**For now, we have to switch from swap-scratch to more preferable swap-move by deleting the scratch slot.**

BTW: I have briefly run through Zephyr boards and all are not optimal, they have equal sizes or one sector difference recommended by the MCUboot documentation. This is the place for improvement.

Reply:
> BTW: I have briefly run through Zephyr boards and all are not optimal, they have equal sizes or one sector difference recommended by the MCUboot documentation. This is the place for improvement.

Yes they should be improved, the debug log is a new feature so existing boards haven't really been updated.